### PR TITLE
[#4089] fix(hive catalog): the problem of slow acquisition of hive table list

### PR DESCRIPTION
**Pull request by mygrsun**
### What changes were proposed in this pull request?

the problem of slow acquisition of hive table list.
Using listTableNamesByFilter replace the getTableObjectsByName method.

### Why are the changes needed?

I found that list-table will takes 300s when a schema has 5000 tables .

Fix: #4089 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

Manual testing

### CODE REVIEW

File: catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java [hunk 1]
```diff
@@ -557,23 +556,18 @@ public NameIdentifier[] listTables(Namespace namespace) throws NoSuchSchemaExcep
...
-                          return isHiveTable(parameters);
-                        }
-                        return true;
-                      })
-                  .map(tb -> NameIdentifier.of(namespace, tb.getTableName()))
-                  .toArray(NameIdentifier[]::new));
+      if (!listAllTables) {
+        String filter =
+            String.format(
+                "%stable_type = \"ICEBERG\"", hive_metastoreConstants.HIVE_FILTER_FIELD_PARAMS);
```
Comment (mchades) [inline catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java:562]: Filtering out only the Iceberg table seems to be insufficient, and we currently also do not support external and view types.
So I think you should use show managed type table when `listAllTables=false`

**Comment by mchades** [comment]
Hi, @mygrsun could you plz resolve the comment and CI issue? Thanks

File: catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java (hunk 1 above)
Comment (mygrsun) [reply catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java:562]: I found the listTableNamesByFilter can’t support filter by table type.so , this method can't meet the requirement of filting view tables.
image

Comment (mchades) [reply catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java:562]: If we do not filter the view, what impact will it have on Gravitino?

Comment (mygrsun) [reply catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java:566]: > If we do not filter the view, what impact will it have on Gravitino?

I have tested it.The view table ,don‘t have any impact ,so i think we can do like this.

Comment (mygrsun) [reply catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java:566]: I add some comment ,please review again

**Comment by mchades** [comment]
@FANNG1 @yuqi1129 Can you help review it?

File: catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java (hunk 1 above)
Comment (mygrsun) [reply catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java:567]: Now,the ci issue is a troublesome problem. https://github.com/apache/gravitino/actions/runs/10452698362/job/28942732918?pr=4469
 This ci issue is due to a bug in hive, and the iceberg project has the same proble.
https://github.com/apache/iceberg/pull/2722#issuecomment-867363019
image

This hive bug is only happen in Derby. In our environment , the metastore storage is mysql,so we don't encounter this problem.

**Comment by mygrsun** [comment]
Now,the ci issue is a troublesome problem. https://github.com/apache/gravitino/actions/runs/10452698362/job/28942732918?pr=4469
This ci issue is due to a bug in hive, and the iceberg project has the same proble.
https://github.com/apache/iceberg/pull/2722#issuecomment-867363019
image

This hive bug is only happen in Derby. In our environment , the metastore storage is mysql,so we don't encounter this problem.

**Comment by mchades** [comment]
@mygrsun Is there any progress? May I take on this?

**Comment by mygrsun** [comment]
> @mygrsun Is there any progress? May I take on this?

thanks,you can do it

**Comment by mchades** [comment]
I have verified locally that through this PR, the time consumption of listing 1000 tables can be reduced from **2043ms** to **14ms**.

It's ready for review now. @jerryshao

File: catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java [hunk 2]
```diff
@@ -539,23 +538,32 @@ public NameIdentifier[] listTables(Namespace namespace) throws NoSuchSchemaExcep
...
-                        }
-                        if (!listAllTables) {
-                          Map<String, String> parameters = tb.getParameters();
-                          return isHiveTable(parameters);
-                        }
-                        return true;
-                      })
-                  .map(tb -> NameIdentifier.of(namespace, tb.getTableName()))
-                  .toArray(NameIdentifier[]::new));
+      if (!listAllTables) {
```
Comment (jerryshao) [inline catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java:541]: @mchades I think you should also check Paimon catalog, AFAIK, I have supported Paimon with HMS backend recently. Please sync with @FANNG1 and @caican00 .

Comment (mchades) [reply catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java:541]: Based on [the source codes of Paimon ](https://github.com/apache/paimon/blob/76f906faad4a155f75f6db52a5c0014d532b2c55/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java#L1021)and tested locally, I updated the PR and use table param `table_type=paimon` to filter out Paimon tables.

File: catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java [hunk 3]
```diff
@@ -539,23 +538,32 @@ public NameIdentifier[] listTables(Namespace namespace) throws NoSuchSchemaExcep
...
+        // The reason for using the listTableNamesByFilter function is that the
+        // getTableObjectiesByName function has poor performance. Currently, we focus on the
+        // Iceberg, Paimon and Hudi table. In the future, if necessary, we will need to filter out
+        // other tables. In addition, the current return also includes tables of type VIRTUAL-VIEW.
+        String icebergAndPaimonFilter = getIcebergAndPaimonFilter();
+        List<String> icebergAndPaimonTables =
+            clientPool.run(
+                c ->
+                    c.listTableNamesByFilter(
+                        schemaIdent.name(), icebergAndPaimonFilter, (short) -1));
```
Comment (jerryshao) [inline catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java:551]: What's the meaning of `(short) -1` here? Can you define a constant and add a comment on it?

Comment (mchades) [reply catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java]: fixed

**Review by jerryshao — APPROVED** [review]
LGTM.

**Pull request merged**